spec/inputs/heartbeat_spec.rb

```diff
@@ -1 +1,92 @@
 require "logstash/devutils/rspec/spec_helper"
+require "logstash/inputs/heartbeat"
+
+def fetch_event(config, number)
+  pipeline = LogStash::Pipeline.new(config)
+  queue = Queue.new
+  pipeline.instance_eval do
+    @output_func = lambda { |event| queue << event }
+  end
+  pipeline_thread = Thread.new { pipeline.run }
+  counter = 0
+  begin
+    event = queue.pop
+    pipeline_thread.join
+    counter += 1
+  end until counter == number
+  return event
+end
+
+describe LogStash::Inputs::Heartbeat do
+  context "Default message test" do
+    subject do
+      next LogStash::Inputs::Heartbeat.new({})
+    end
+
+    it "should generate an 'ok' message" do
+      sequence = 1
+      subject.generate_message(sequence) do |event|
+        insist { event['message'] } == 'ok'
+      end
+    end # it "should generate an 'ok' message"
+  end # context "Default message test"
+
+  context "Simple message test" do
+    subject do
+      next LogStash::Inputs::Heartbeat.new({"message" => "my_message"})
+    end
+
+    it "should generate a message containing 'my_message'" do
+      sequence = 2
+      subject.generate_message(sequence) do |event|
```

Comment:
I'm going to set it only once at the beginning of the class now, to clear that up.